Flutter/datagrid/grouping.md

```diff
@@ -0,0 +1,1155 @@
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('Designation',
+                          overflow: TextOverflow.ellipsis))),
+              GridColumn(
+                  columnName: 'Salary',
+                  label: Container(
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('Salary'))),
+            ]),
+      ),
+    );
+  }
+        
+{% endhighlight %}
+{% endtabs %}
+
+## Customize group expander icon
+
+The SfDataGrid allows you to change the group expander icon by using the `SfDataGridThemeData.groupExpandIconData` property. The DataGrid should be wrapped inside the SfDataGridTheme.
+
+The `SfDataGridThemeData` and `SfDataGridTheme` classes are available in the syncfusion_flutter_core package. So, import the following file.
+
+{% tabs %}
+{% highlight Dart %}
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Syncfusion Flutter DataGrid')),
+      body: SfDataGridTheme(
+        data: SfDataGridThemeData(
+            groupExpanderIcon: Icon(Icons.expand_circle_down_outlined,
+                size: 20, color: Colors.pink)),
+        child: SfDataGrid(
+            source: employeeDataSource,
+            allowExpandCollapseGroup: true,
+            columns: <GridColumn>[
+              GridColumn(
+                  columnName: 'ID',
+                  label: Container(
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('ID'))),
+              GridColumn(
+                  columnName: 'Name',
+                  label: Container(
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('Name'))),
+              GridColumn(
+                  columnName: 'Designation',
+                  label: Container(
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('Designation',
+                          overflow: TextOverflow.ellipsis))),
+              GridColumn(
+                  columnName: 'Salary',
+                  label: Container(
+                      padding: EdgeInsets.all(8),
+                      alignment: Alignment.center,
+                      child: Text('Salary'))),
+            ]),
+      ),
+    );
+  }
+        
+{% endhighlight %}
+{% endtabs %}
+
+## Limitations
+
+* When conducting CRUD operations on rows through actions like `editing,` `pull to refresh,` `load more,` or directly on the data source, and when CRUD operations are performed on the columns collection, it triggers a refresh in the grouping. Consequently, this action resets the expansion and collapse states based on the `SfDataGrid.autoExpandGroups` property.
```

Comment:
* The grouping will be refreshed when performing CRUD operations on the `DataGridSource.rows` and the `SfDataGrid.columns`. Consequently, this action resets the expand and collapse states of the groups based on the `SfDataGrid.autoExpandGroups` property.
* To prevent unnecessary grouping refresh, ensure that the `SfDataGrid.columns` property is set as an instance.

Reply:
Modified.